libs/dal/src/types/error.enum.ts

```diff
@@ -0,0 +1,3 @@
+export enum ErrorCodesEnum {
+  DUPLICATE_KEY = '11000',
+}
```

Comment:
I made this enum as a temporary solution. It helps the application layer recognize errors from the data access layer without being dependent on MongoDB. But I'm considering whether it's better to generate our own error codes and link them to the application layer for better control and clarity.